test/gtest/ucp/test_ucp_tag_offload.cc

```diff
@@ -469,6 +486,37 @@ UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_tag_offload_selection, self_rcx,
                               "self,rc_x");
 
 
+class test_ucp_tag_offload_cuda : public test_ucp_tag_offload {
+public:
+    test_ucp_tag_offload_cuda() {
+        modify_config("RNDV_THRESH", "1024");
+    }
+};
+
+// Test that expected SW RNDV request is handled properly when receive buffer
+// is allocated on GPU memory.
+UCS_TEST_P(test_ucp_tag_offload_cuda, sw_rndv_to_cuda_mem, "TM_SW_RNDV=y")
+{
+    activate_offload(sender());
+
+    size_t size   = 2048;
+    ucp_tag_t tag = 0xCAFEBABE;
+    // Test will be skipped here if CUDA mem is not supported
+    void *rbuf    = mem_buffer::allocate(size, UCS_MEMORY_TYPE_CUDA);
+    request *rreq = recv_nb_exp(rbuf, size, DATATYPE, tag, UCP_TAG_MASK_FULL);
+
+    std::vector<uint8_t> sendbuf(size); // can send from any memory
+    request *sreq = (request*)ucp_tag_send_nb(sender().ep(), &sendbuf[0],
+                                              size, DATATYPE, tag,
+                                              send_callback);
+    wait_and_validate(rreq);
+    wait_and_validate(sreq);
+}
```

Comment:
should do:
```C
mem_buffer::release(rbuf, UCS_MEMORY_TYPE_CUDA);
```
at the end of the testing

or just use:
```C
mem_buffer rbuf(size, UCS_MEMORY_TYPE_CUDA);
```
and then there is no need to explicitly release memory buffer